solr/solrj/src/java/org/apache/solr/common/cloud/PerReplicaStates.java

```diff
@@ -130,6 +140,26 @@ public State get(String replica) {
     return states.get(replica);
   }
 
+  /**
+   * A new child node got created. construct a new Object with this
+   *
+   * @param newState the newly inserted state
+   * @param czxid create tx id of the child node
+   */
+  public PerReplicaStates insert(State newState, long czxid) {
+    State existing = states.get(newState.replica);
+    if (existing == null || existing.version < newState.version) {
+      LinkedHashMap<String, State> copy = new LinkedHashMap<>(states.size());
+      states.forEachEntry(copy::put);
+      copy.put(newState.replica, newState);
+      return new PerReplicaStates(
+          path,
+          new WrappedSimpleMap<>(copy),
+          czxid); // child node's czxid is same as state.json pzxid
+    }
+    return null;
```

Comment:
Maybe good to include in javadoc that if the insertion is not successful, then `null` will be returned, as such is not that obvious to the caller